### PR TITLE
Add sample rate to SDK docs.

### PR DESCRIPTION
**Pull request by bretthoerner**
This is short and sweet, I didn't know what else to say.

### CODE REVIEW

File: docs/clientdev/overview.rst [hunk 1]
```diff
@@ -132,6 +133,20 @@ The resulting POST request would then transmit to::
...
 
+Event sampling
+--------------
+
+SDKs should allow the user to configure what percentage of events are actually
+sent to the server (the rest should be silently ignored). For example:
+
+.. sourcecode:: python
+
+    sample_rate = options.get('sample_rate') or 1.0
```
Comment (LewisJEllis) [inline docs/clientdev/overview.rst]: if `options.get('sample_rate')` returns 0 this expression will evaluate to 1.0 instead, i think theres a legitimate case for setting sample_rate to 0?

File: docs/clientdev/overview.rst [hunk 2]
```diff
@@ -132,6 +133,20 @@ The resulting POST request would then transmit to::
...
+
+    # assuming random() returns a value between 0.0 and 1.0
```
Comment (LewisJEllis) [inline docs/clientdev/overview.rst]: maybe clarify 0 inclusive 1.0 exclusive, details might technically matter as ted pointed out in https://github.com/getsentry/raven-python/pull/978

File: docs/clientdev/overview.rst (hunk 1 above)
Comment (bretthoerner) [inline docs/clientdev/overview.rst]: Welp, you're the CEO now.

File: docs/clientdev/overview.rst (hunk 2 above)
Comment (bretthoerner) [inline docs/clientdev/overview.rst]: Good call, will do.

**Pull request merged**